### PR TITLE
feat(analytics): initiate on device measurement with sha256-hashed values

### PR DESCRIPTION
**Pull request by matinzd**
### Description

Add support for on-device measurement hashed values

### Related issues

N/A

### Release Summary

Added support for on-device measurement initiation with hashed phone number or email address.

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [x] `iOS`
- My change includes tests;
  - [x] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [x] `jest` tests added or updated in `packages/\*\*/__tests__`
- [x] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [x] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Comment by matinzd** [comment]
I think this is ready :) @russellwheatley @Lyokone

File: packages/analytics/__tests__/analytics.test.ts [hunk 1]
```diff
@@ -843,10 +877,18 @@ describe('Analytics', function () {
       expect(initiateOnDeviceConversionMeasurementWithEmailAddress).toBeDefined();
     });
 
+    it('`initiateOnDeviceConversionMeasurementWithHashedEmailAddress` function is properly exposed to end user', function () {
```
Comment (russellwheatley) [inline packages/analytics/__tests__/analytics.test.ts:880]: Do you mind just adding the same unit test you've done above for modular API as well? 

e.g.
```js
 expect(() => initiateOnDeviceConversionMeasurementWithHashedEmailAddress(analytics, '+1234567890');
      ).toThrowError(
        "firebase.analytics().initiateOnDeviceConversionMeasurementWithHashedPhoneNumber(*) 'hashedPhoneNumber' expected a sha256-hashed value of a phone number in E.164 format.",
      );
    });

```

**Review by russellwheatley — CHANGES_REQUESTED** [review]
@matinzd - this is excellent! Well done 👍 . I left some small update which will help maintain test coverage, but once that is done, this LGTM,

File: packages/analytics/__tests__/analytics.test.ts (hunk 1 above)
Comment (matinzd) [reply packages/analytics/__tests__/analytics.test.ts:880]: Done and rebased! [830cec4](https://github.com/invertase/react-native-firebase/pull/7963/commits/830cec42e9f77643e80f1ade4efffd4335c39663)

**Comment by matinzd** [comment]
@russellwheatley As a reminder, this is ready.

File: packages/analytics/__tests__/analytics.test.ts [hunk 2]
```diff
@@ -843,10 +877,26 @@ describe('Analytics', function () {
...
+    });
+
+    it('`initiateOnDeviceConversionMeasurementWithHashedPhoneNumber` should throw if the value is in E.164 format', function () {
+      expect(() =>
+        initiateOnDeviceConversionMeasurementWithHashedPhoneNumber(getAnalytics(), '+1234567890'),
+      ).toThrowError(
+        "firebase.analytics().initiateOnDeviceConversionMeasurementWithHashedPhoneNumber(*) 'hashedPhoneNumber' expected a sha256-hashed value of a phone number in E.164 format.",
+      );
+    });
+
```
Comment (russellwheatley) [inline packages/analytics/__tests__/analytics.test.ts:891]: @matinzd - Could you also write modular test for this one as well, please?

```js
it('throws if not a string', function () {
      expect(() =>
        // @ts-ignore
        firebase.analytics().initiateOnDeviceConversionMeasurementWithHashedEmailAddress(true),
      ).toThrowError(
        "firebase.analytics().initiateOnDeviceConversionMeasurementWithHashedEmailAddress(*) 'hashedEmailAddress' expected a string value.",
      );
    });
```

That should be it I think 😄

**Review by russellwheatley — CHANGES_REQUESTED** [review]

File: packages/analytics/__tests__/analytics.test.ts (hunk 2 above)
Comment (matinzd) [reply packages/analytics/__tests__/analytics.test.ts:900]: Done :D

File: packages/analytics/__tests__/analytics.test.ts [hunk 3]
```diff
@@ -843,10 +877,35 @@ describe('Analytics', function () {
...
     });
 
+    it('`initiateOnDeviceConversionMeasurementWithHashedEmailAddress` function is properly exposed to end user', function () {
+      expect(initiateOnDeviceConversionMeasurementWithHashedEmailAddress).toBeDefined();
+    });
+
+    it('`initiateOnDeviceConversionMeasurementWithHashedEmailAddress` throws if not a string', function () {
+      expect(() =>
+        // @ts-ignore
+        firebase.analytics().initiateOnDeviceConversionMeasurementWithHashedEmailAddress(true),
```
Comment (russellwheatley) [inline packages/analytics/__tests__/analytics.test.ts:887]: @matinzd - I mean modular API like this:
```suggestion
	initiateOnDeviceConversionMeasurementWithHashedEmailAddress(getAnalytics(), true),
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]

File: packages/analytics/__tests__/analytics.test.ts (hunk 3 above)
Comment (matinzd) [reply packages/analytics/__tests__/analytics.test.ts]: Yeah sorry :) Done.

**Review by russellwheatley — APPROVED** [review]

**Comment by matinzd** [comment]
Rebased.

**Review by Lyokone — APPROVED** [review]

**Comment by matinzd** [comment]
Any plans to release this? @russellwheatley @Lyokone

**Pull request merged**